sources/packages/backend/apps/queue-consumers/src/processors/virus-scan/_tests_/virus-scan.processor.e2e-spec.ts

```diff
@@ -40,12 +46,147 @@ describe(describeProcessorRootTest(QueueNames.FileVirusScanProcessor), () => {
+      uniqueFileName: studentFile.uniqueFileName,
+      fileName: studentFile.fileName,
+    });
+
+    // Act
+    const errorMessage = `Unable to scan the file ${studentFile.uniqueFileName} for viruses. Connection to ClamAV server failed.`;
+    await expect(
+      processor.performVirusScan(mockedJob.job),
+    ).rejects.toStrictEqual(new Error(errorMessage));
+    expect(
+      mockedJob.containLogMessages([
+        "Log details",
+        "Starting virus scan.",
+        "ClamAV server availability: available.",
+        errorMessage,
+      ]),
+    ).toBe(true);
+  });
+
+  it("Should throw an error when the virus scan server is unavailable.", async () => {
+    // Arrange
+    const studentFile = createFakeStudentFileUpload();
+    studentFile.virusScanStatus = VirusScanStatus.InProgress;
+    await db.studentFile.save(studentFile);
+    clamAVServiceMock.scanFile = jest.fn(() =>
+      Promise.resolve({
+        isInfected: null,
+        errorCode: SERVER_UNAVAILABLE,
+      }),
+    ); // Queued job.
+    const mockedJob = mockBullJob<VirusScanQueueInDTO>({
+      uniqueFileName: studentFile.uniqueFileName,
+      fileName: studentFile.fileName,
+    });
+
+    // Act
+    const errorMessage = `Unable to scan the file ${studentFile.uniqueFileName} for viruses. ClamAV server is unavailable.`;
+    await expect(
+      processor.performVirusScan(mockedJob.job),
+    ).rejects.toStrictEqual(new Error(errorMessage));
+    expect(
+      mockedJob.containLogMessages([
+        "Log details",
+        "Starting virus scan.",
+        "ClamAV server availability: unavailable.",
+        errorMessage,
+      ]),
+    ).toBe(true);
+  });
+
+  it("Should throw an error when an unknown error occurred during virus scanning.", async () => {
+    // Arrange
+    const studentFile = createFakeStudentFileUpload();
+    studentFile.virusScanStatus = VirusScanStatus.InProgress;
+    await db.studentFile.save(studentFile);
+    clamAVServiceMock.scanFile = jest.fn(() =>
+      Promise.resolve({
+        isInfected: undefined,
+        errorCode: UNKNOWN_ERROR,
+      }),
+    ); // Queued job.
+    const mockedJob = mockBullJob<VirusScanQueueInDTO>({
+      uniqueFileName: studentFile.uniqueFileName,
+      fileName: studentFile.fileName,
+    });
+
+    // Act
+    const errorMessage = `Unable to scan the file ${studentFile.uniqueFileName} for viruses. Unknown error.`;
+    await expect(
+      processor.performVirusScan(mockedJob.job),
+    ).rejects.toStrictEqual(new Error(errorMessage));
+    expect(
+      mockedJob.containLogMessages([
+        "Log details",
+        "Starting virus scan.",
+        "ClamAV server availability: uncertain.",
+        errorMessage,
+      ]),
+    ).toBe(true);
+    const scannedStudentFile = await db.studentFile.findOneBy({
+      id: studentFile.id,
+    });
+    expect(scannedStudentFile.virusScanStatus).toBe(VirusScanStatus.InProgress);
+    expect(scannedStudentFile.fileName).toBe(studentFile.fileName);
```

Comment:
Can we add this for the other tests that are verifying errors except for the test where file is not found.

Reply:
This is added for those test cases.